lib/internal/modules/esm/module_job.js

```diff
@@ -393,8 +393,33 @@ class ModuleJobSync extends ModuleJobBase {
       throw new ERR_REQUIRE_ASYNC_MODULE();
     }
     setHasStartedUserESMExecution();
-    const namespace = this.module.evaluateSync();
-    return { __proto__: null, module: this.module, namespace };
+    try {
+      const namespace = this.module.evaluateSync();
+      return { __proto__: null, module: this.module, namespace };
+    } catch (e) {
+      if (e?.name === 'ReferenceError' &&
```

Comment:
Instead of copying, can you abstract the error message modification code into a helper and reuse it here? Otherwise it can get easily out of sync.